internal/proto/reader.go

```diff
@@ -28,20 +46,34 @@ func (RedisError) RedisError() {}
 
 //------------------------------------------------------------------------------
 
-type MultiBulkParse func(*Reader, int64) (interface{}, error)
+var (
+	uvinf    = math.Inf(1)
+	uvneginf = math.Inf(-1)
+)
+
+type (
+	AggregateBulkParse func(*Reader, byte, int64) (interface{}, error)
+	MultiBulkParse     func(*Reader, int64) (interface{}, error)
+)
```

Comment:
Ideally we should not introduce `AggregateBulkParse` - see my comment about `ReadMapReply`.